pkg/apis/duck/v1alpha1/identity.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+const (
+	WorkloadIdentityFailed        = "WorkloadIdentityReconcileFailed"
+	WorkloadIdentitySucceed       = "WorkloadIdentityReconciled"
+	DeleteWorkloadIdentityFailed  = "WorkloadIdentityDeleteFailed"
+	DeleteWorkloadIdentitySucceed = "WorkloadIdentityDeleted"
+)
+
+// WorkloadIdentityStatus represents the status of WorkloadIdentity.
+type WorkloadIdentityStatus struct {
+	// Enabled represents whether workload identity is enabled or not.
+	Enabled string `json:"enabled,omitempty"`
+
+	// Ready represents whether workload identity reconciles successfully or not. If it is not enabled, Ready will be unknown.
+	Ready string `json:"ready,omitempty"`
+
+	// Short reason for the status.
+	Reason string `json:"reason,omitempty"`
+
+	// A human readable message indicating details about the failure.
+	Message string `json:"message,omitempty"`
```

Comment:
it seems that we should be able to reuse Condition from pkg for these?
See https://github.com/knative/pkg/blob/0840da9555a3a75f801abc1d654fb00dfe9a687a/apis/condition_types.go#L58?
And you will get more type-safety.
The duckv1.SourceStatus has a Status, which has a list of Conditions. 
I'm not entirely sure we should be adding these ones here...

Reply:
I think I'd rather keep it simple for now to see how this workload identity places out. 
And would just add a condition to the types called something like "WorkloadIdentityConfigured" or something like that, not part of the NewLivingConditionSet...
and provide methods to MarkWorkloadIdentityConfigured and not, with proper reasons....
We can use the Unknown when users don't specify a serviceAccount... nothing will be set, but that won't affect the readiness.
We can use True when users specified the serviceAccount and it configured correctly.
We can use False when users specified the serviceAccount and it wasn't configured properly. In this latter case, we should also Mark Ready = false to the object.

I don't think we need to save Enabled in status. But it would be nice if we save the k8s service account, what you are currently doing. 

I think this will simplify things quite a bit... but want to hear your thoughts as you have done all the work here.

Reply:
I'll change that, and save the k8s service account name. I'll combine the new commit with change of k8s name.

Reply:
It seems I can only set the condition in every reconciler, rather than combine it to the identity reconciler if we use the Conditions for WI.

The probelm is still for the status. If I want to mark the condition status in the identity reconciler, I need to get sources/channel `status` and `api.condtionSet`. Then I can use `sourceStatus.Manager(api.condtionSet).MarkTrue` to mark the condition.

I can get the condSet, but I cannot create an identifiable interface function for getting all sources and channel’s status, the only common thing shared by sources and channel’s status is they all use duck.status. Then I have to write things like (s *duck.status)MarkworkloadIdentityReconciled. I am not sure if we can do that.